src/js/components/Tab/Tab.js

```diff
@@ -12,194 +12,184 @@ import { normalizeColor } from '../../utils';
-        });
+        borderColor = theme.tab.border.active.color || borderColor;
+      } else if (disabled && theme.tab.border.disabled) {
+        borderColor = theme.tab.border.disabled.color || borderColor;
+      } else if (over) {
+        borderColor = theme.tab.border.hover.color || borderColor;
       }
-      if (disabled) {
-        return React.cloneElement(iconProp, { ...theme.tab.disabled });
-      }
-      return React.cloneElement(iconProp, {
-        color: over ? theme.tab.hover.color : theme.tab.color,
-      });
-    };
+      borderColor = normalizeColor(borderColor, theme);
 
-    let normalizedIcon;
-    if (icon) {
-      normalizedIcon = renderIcon(icon);
+      tabStyles.border = {
+        side: theme.tab.border.side,
+        size: theme.tab.border.size,
+        color: borderColor,
+      };
     }
 
-    const first = reverse ? normalizedTitle : normalizedIcon;
-    const second = reverse ? normalizedIcon : normalizedTitle;
+    tabStyles.background = active
+      ? theme.tab.active.background || theme.tab.background
+      : theme.tab.background;
+    tabStyles.pad = theme.tab.pad;
+    tabStyles.margin = theme.tab.margin;
+  }
 
-    let withIconStyles;
-    if (first && second) {
-      withIconStyles = {
-        direction: 'row',
-        align: 'center',
-        justify: 'center',
-        gap: 'small',
-      };
+  // needed to apply hover/active styles to the icon
+  const renderIcon = (iconProp) => {
+    if (active) {
+      return React.cloneElement(iconProp, {
+        ...theme.tab.active,
+      });
     }
-
-    return (
-      <Button
-        ref={ref}
-        plain
-        role="tab"
-        aria-selected={active}
-        aria-expanded={active}
+    if (disabled) {
+      return React.cloneElement(iconProp, { ...theme.tab.disabled });
+    }
+    return React.cloneElement(iconProp, {
+      color: over ? theme.tab.hover.color : theme.tab.color,
+    });
+  };
+
+  let normalizedIcon;
+  if (icon) {
+    normalizedIcon = renderIcon(icon);
+  }
+
+  const first = reverse ? normalizedTitle : normalizedIcon;
+  const second = reverse ? normalizedIcon : normalizedTitle;
+
+  let withIconStyles;
+  if (first && second) {
+    withIconStyles = {
+      direction: 'row',
+      align: 'center',
+      justify: 'center',
+      gap: 'small',
+    };
+  }
+
+  return (
+    <Button
+      ref={ref}
+      plain
+      role="tab"
+      aria-selected={active}
+      aria-expanded={active}
+      disabled={disabled}
+      {...rest}
+      onClick={onClickTab}
+      onMouseOver={onMouseOverTab}
+      onMouseOut={onMouseOutTab}
+      onFocus={() => {
+        setFocus(true);
+        if (onMouseOver) onMouseOver();
+      }}
+      onBlur={() => {
+        setFocus(undefined);
+        if (onMouseOut) onMouseOut();
+      }}
+      // ensure focus outline is not covered by hover styling
+      // of adjacent tabs
+      style={focus && { zIndex: 1 }}
+      tabIndex={active ? 0 : -1}
```

Comment:
Nice w roving tab index. Wondering if this will have any effect on people depending on how they've written e2e tests. That being said, I think we should consider the existing behavior a "bug" since it didn't follow existing accessibility best practices so we should be fine wrt backwards compatibility.

Reply:
Also, do we need to pass through `0` or will a button automatically be in the Tab focus? Not sure but could we do `tabIndex={!active && -1}` or something like that? Haven't tried locally -- was just trying to avoid adding more to the DOM if tab-index=0 is redundant

Reply:
I could change this to `tabIndex={active ? undefined : -1}`.

`tabIndex={!active && -1}` or variations of this attempt to pass a boolean as the value.

Reply:
> Nice w roving tab index. Wondering if this will have any effect on people depending on how they've written e2e tests. That being said, I think we should consider the existing behavior a "bug" since it didn't follow existing accessibility best practices so we should be fine wrt backwards compatibility.

Yeah this change could definitely cause some failing tests since the keyboard behavior is changing. I agree that it is a fix, just need to make it clear in the release notes that the keyboard behavior of Tabs is changing